### PR TITLE
Feat/governance update

### PR DESCRIPTION
**Pull request by IanAtCredo**

### CODE REVIEW

File: credoai/governance/governance.py [hunk 1]
```diff
@@ -270,19 +273,25 @@ def _check_inclusion(self, label, evidence):
...
+        meta = {"client": "Credo AI Lens", "version": __version__}
+        data = json_dumps(serialize(data=data, meta=meta))
+        with open(filename, "w") as f:
+            f.write(data)
+
+    def _prepare_export_data(self):
+        evidences = self._prepare_evidences()
         data = {
             "policy_pack_id": self._policy_pack_id,
+            "models": [self._model] if self._model else None,
```
Comment (IanAtCredo) [inline credoai/governance/governance.py:290]: This is the important new thing

File: credoai/lens/lens.py [hunk 2]
```diff
@@ -184,6 +183,8 @@ def send_to_governance(self, overwrite_governance=False):
         """
         evidence = self.get_evidence()
         if self.gov:
+            if self.model:
```
Comment (IanAtCredo) [inline credoai/lens/lens.py:186]: If model exists, send information about it. Not sure what to do when no model is provided!

File: credoai/governance/governance.py [hunk 3]
```diff
@@ -250,11 +220,79 @@ def export(self, filename=None):
...
+            prepared_model["assessment_dataset_name"] = assessment_dataset.name
+        self._model = prepared_model
+
+    def set_evidence(self, evidences: List[Evidence]):
+        """
+        Update evidences
+        """
+        self._evidences = evidences
+
+    def tag_model(self, model):
```
Comment (fabrizio-credo) [inline credoai/governance/governance.py:275]: Can you add a docstring for this method?

> It has a lot of print statements, I think we should change them to `global_logger.info`

Never mind I didn't see it was interactive.

File: credoai/governance/governance.py [hunk 4]
```diff
@@ -250,11 +220,79 @@ def export(self, filename=None):
         global_logger.info(export_status)
         return to_return
 
-    def _api_export(self, evidences):
+    def get_evidence_requirements(self, tags: dict = None):
```
Comment (fabrizio-credo) [inline credoai/governance/governance.py:223]: Update docstring: only evidences with tags matching model tags, or with no tags are returned

File: credoai/governance/governance.py [hunk 5]
```diff
@@ -250,11 +220,79 @@ def export(self, filename=None):
...
+        return self._unique_tags
+
+    def get_model_tags(self):
+        """Get the tags for the associated model"""
+        if self._model:
+            return self._model["tags"]
+        else:
+            return None
+
+    def set_artifacts(self, model, training_dataset=None, assessment_dataset=None):
```
Comment (fabrizio-credo) [inline credoai/governance/governance.py:258]: Adding docstring here would be useful.

**Review by fabrizio-credo — APPROVED** [review]
Looks good to me and aligned to Chang's specs.

In general I would add docstrings, in particular for the methods I commented.

**Pull request merged**